lib/pure/options.nim

```diff
@@ -34,35 +36,37 @@ runnableExamples:
   let found = "abc".find('c')
   assert found.isSome and found.get() == 2
 
-## The `get` operation demonstrated above returns the underlying value, or
-## raises `UnpackDefect` if there is no value. Note that `UnpackDefect`
-## inherits from `system.Defect` and should therefore never be caught.
-## Instead, rely on checking if the option contains a value with the
-## `isSome <#isSome,Option[T]>`_ and `isNone <#isNone,Option[T]>`_ procs.
-##
-##
-## Pattern matching
-## ================
-##
-## .. note:: This requires the [fusion](https://github.com/nim-lang/fusion) package.
-##
-## [fusion/matching](https://nim-lang.github.io/fusion/src/fusion/matching.html)
-## supports pattern matching on `Option`s, with the `Some(<pattern>)` and
-## `None()` patterns.
-##
-## .. code-block:: nim
-##   {.experimental: "caseStmtMacros".}
-##
-##   import fusion/matching
-##
-##   case some(42)
-##   of Some(@a):
-##     assert a == 42
-##   of None():
-##     assert false
-##
-##   assertMatch(Some(Some(None())), some(some(none(int))))
-# xxx pending https://github.com/timotheecour/Nim/issues/376 use `whichModule`
+##[
+The `get` operation demonstrated above returns the underlying value, or
+raises `UnpackDefect` if there is no value. Note that `UnpackDefect`
+inherits from `system.Defect` and should therefore never be caught.
+Instead, rely on checking if the option contains a value with the
+`isSome <#isSome,Option[T]>`_ and `isNone <#isNone,Option[T]>`_ procs.
+
+
+Pattern matching
+================
+
+.. note:: This requires the [fusion](https://github.com/nim-lang/fusion) package.
+
+[fusion/matching](https://nim-lang.github.io/fusion/src/fusion/matching.html)
+supports pattern matching on `Option`s, with the `Some(<pattern>)` and
+`None()` patterns.
+
+.. code-block:: nim
+  {.experimental: "caseStmtMacros".}
+
+  import fusion/matching
+
+  case some(42)
+  of Some(@a):
+    assert a == 42
+  of None():
+    assert false
+
+  assertMatch(Some(Some(None())), some(some(none(int))))
```

Comment:
- [ ] `assertMatch(Some(Some(None())), some(some(none(int))))` doesn't work:

Error: unhandled exception: Expected one of {nnkExprColonExpr} but node has kind nnkCall (assertion on (filename: "matching.nim", line: 857, column: 12)) [AssertionDefect]

`Some(Some(None())) := some(some(none(int)))` does work but would need a comment explaning what `:=` means

- [ ] the error message above is bad, should probably use `toLocation` from `lib/std/private/miscdollars.nim` instead (even if it's private; IMO fusion should be able to use things under `lib/std/private/` as it's intended to work in tandem with nim)

Reply:
Oh yeah, I swapped the arguments, it should be `assertMatch(some(some(none(int))), Some(Some(None())))`. Do you have any preference on `:=` vs `assertMatch`?

Reply:
I prefer `assertMatch` (i generally dislike non-standard operators)